### PR TITLE
python benchmark infra for jit kernel speed

### PR DESCRIPTION
**Pull request by FDecaYed**
This PR added a nvprof based kernel benchmark method in python unit test.
With this, user can now easily write python test to compare jit and native kernel speed on various config and sizes.
I'm using it for testing kernel speed without effect of fusion considered, but it is also suitable to test fused kernel speed against multiple native kernel since the parser aggregates kernel time.

```
root@6e328e4ca2b3:/jit_code# CUDA_FUSER_BENCHMARK=1 python pytorch/test/test_jit_cuda_fuser.py TestCudaFuser.test_benchmark_all
Benchmark not meeting threshold of 1.2: ['batchnorm', '34', '1024', '14', '14', 'float16', 'jit', 18777.7, 'native', 112.50377358490564]
Benchmark not meeting threshold of 1.2: ['batchnorm', '34', '128', '14', '14', 'float32', 'jit', 13589.9, 'native', 24.455150665732308]
Benchmark not meeting threshold of 1.2: ['layernorm', '32', '36', '30', '40', 'normshape', '30', '40', 'float16', 'jit', 35.477, 'native', 16.15524034672971]
Benchmark not meeting threshold of 1.2: ['layernorm', '32', '36', '30', '40', 'normshape', '30', '40', 'float32', 'jit', 36.462, 'native', 27.177255047230968]
Benchmark not meeting threshold of 1.2: ['softmax', '256', '128', 'float16', 'jit', 3.7571, 'native', 1.8329]
Benchmark not meeting threshold of 1.2: ['softmax', '256', '128', 'float32', 'jit', 3.2647, 'native', 1.6879]
Benchmark not meeting threshold of 1.2: ['softmax', '256', '2048', 'float16', 'jit', 12.385, 'native', 5.4366]
Benchmark not meeting threshold of 1.2: ['softmax', '256', '2048', 'float32', 'jit', 14.238, 'native', 6.2442]
Benchmark not meeting threshold of 1.2: ['softmax', '256', '32', 'float16', 'jit', 2.5125, 'native', 1.5068]
Benchmark not meeting threshold of 1.2: ['softmax', '256', '32', 'float32', 'jit', 2.4673, 'native', 1.4991]
Benchmark not meeting threshold of 1.2: ['softmax', '50000', '128', 'float16', 'jit', 74.608, 'native', 34.249]
Benchmark not meeting threshold of 1.2: ['softmax', '50000', '128', 'float32', 'jit', 88.535, 'native', 63.643]
Benchmark not meeting threshold of 1.2: ['softmax', '50000', '2048', 'float16', 'jit', 1274.28, 'native', 682.77]
Benchmark meeting threshold of 1.2: ['softmax', '50000', '2048', 'float32', 'jit', 1385.83, 'native', 1294.63]
Benchmark not meeting threshold of 1.2: ['softmax', '50000', '32', 'float16', 'jit', 73.414, 'native', 13.153]
Benchmark not meeting threshold of 1.2: ['softmax', '50000', '32', 'float32', 'jit', 73.492, 'native', 18.238]
Benchmark not meeting threshold of 1.2: ['softmax', '8192', '128', 'float16', 'jit', 17.464, 'native', 5.1998]
Benchmark not meeting threshold of 1.2: ['softmax', '8192', '128', 'float32', 'jit', 19.806, 'native', 11.307]
Benchmark not meeting threshold of 1.2: ['softmax', '8192', '2048', 'float16', 'jit', 207.93, 'native', 109.93]
Benchmark meeting threshold of 1.2: ['softmax', '8192', '2048', 'float32', 'jit', 227.82, 'native', 215.29]
Benchmark not meeting threshold of 1.2: ['softmax', '8192', '32', 'float16', 'jit', 16.066, 'native', 3.7499]
Benchmark not meeting threshold of 1.2: ['softmax', '8192', '32', 'float32', 'jit', 16.015, 'native', 3.7437]
.
----------------------------------------------------------------------
Ran 1 test in 93.368s

OK
```

### CODE REVIEW

**Comment by csarofeen** [comment]
We've generally been contributing to the benchmarking done in: https://github.com/csarofeen/pytorch/tree/20_12_3_devel/benchmarks/tensorexpr would it be hard to fit this benchmarking in there? CC @shmsong and @kevinstephano who have both worked on them before.

**Comment by shmsong** [comment]
> We've generally been contributing to the benchmarking done in: https://github.com/csarofeen/pytorch/tree/20_12_3_devel/benchmarks/tensorexpr would it be hard to fit this benchmarking in there? CC @shmsong and @kevinstephano who have both worked on them before.

It can integrate but I guess it'd be easier if the BatchNorm and LayerNorm can be JITed in trace mode instead of script mode. But either way would be possible. I'd just expect some refactoring of the benchmark infra to do this in script mode. @kevinstephano Please comment if you have a straightforward way to do script mode JIT in upstream TE benchmark.

**Comment by csarofeen** [comment]
ping @kevinstephano

**Comment by csarofeen** [comment]
Stale
